nipoppy/workflow/proc_pipe/tractoflow/run_tractoflow.py

```diff
@@ -21,22 +22,59 @@
 
 MEM_MB = 4000
 
-def parse_data(bids_dir, participant_id, session_id, logger=None):
+def parse_data(global_configs, bids_dir, participant_id, session_id, use_bids_filter=False, logger=None):
     """ Parse and verify the input files to build TractoFlow's simplified input to avoid their custom BIDS filter
     """
 
+    ## load from global configs
+    DATASET_ROOT = global_configs["DATASET_ROOT"]
+
     ## because why parse subject ID the same as bids ID?
     subj = participant_id.replace('sub-', '')
 
-    logger.info('Building BIDS Layout...')
+    ## build a regex of anything not subject id
+    srx = re.compile(f"sub-(?!{subj}.*$)")
+
+    logger.info('Building Single Subject BIDS Layout...')
+
+    ## build a BIDSLayoutIndexer to only pull subject ID
+    bidx = BIDSLayoutIndexer(ignore=[srx])
+    
+    ## parse bids directory with subject filter
+    layout = BIDSLayout(bids_dir, indexer=bidx)
+    ## check if DB exists on disk first? BIDSLayout(database_path=var)? where is this saved?
+    ## should this be made / updated as part of BIDS-ification of dicoms?
+
+    ## load the bids filter if it's called
+    if use_bids_filter:
+
+        bidf_path = Path(f"{DATASET_ROOT}", 'proc', 'bids_filter_tractoflow.json') ## is this where it will always be?
```

Comment:
This is where it gets copied from CWD of `run_tractoflow`. I find it a bit annoying to open and edit it outside of the repo and I think it could be helpful to commit these to dataset specific forks.